### PR TITLE
improved lidar recovery

### PR DESCRIPTION
**Pull request by bit-pirate**
- deletes left-over driver instance before initialising
- adds proper disposal of driver instances (pointer magic)
- resets scanning only if driver is healthy

### CODE REVIEW

**Comment by naveedhd** [comment]
@bit-pirate I tested it and it seems to work fine and recover every time. (without the need for driver reinit)

Delivery was launched in the background and it did not break too.

**Comment by bit-pirate** [comment]
:feelsgood:

File: sdk/src/rplidar_driver.cpp [hunk 1]
```diff
@@ -60,16 +60,17 @@ RPlidarDriver * RPlidarDriver::CreateDriver(_u32 drivertype)
 }
 
 
-void RPlidarDriver::DisposeDriver(RPlidarDriver * drv)
+void RPlidarDriver::DisposeDriver(RPlidarDriver ** drv)
 {
-    delete drv;
+  delete *drv;
+  *drv = NULL;
 }
```
Comment (naveedhd) [inline sdk/src/rplidar_driver.cpp]: @bit-pirate why have we used `*drv` instead of `drv`?

File: sdk/src/rplidar_driver.cpp [hunk 2]
```diff
@@ -60,16 +60,17 @@ RPlidarDriver * RPlidarDriver::CreateDriver(_u32 drivertype)
...
 
 
-void RPlidarDriver::DisposeDriver(RPlidarDriver * drv)
+void RPlidarDriver::DisposeDriver(RPlidarDriver ** drv)
 {
-    delete drv;
+  delete *drv;
+  *drv = NULL;
```
Comment (naveedhd) [inline sdk/src/rplidar_driver.cpp]: @bit-pirate why have we used `*drv` instead of `drv`?

**Pull request merged**